app/src/components/market/common/market_top_details_open/index.tsx

```diff
@@ -110,25 +110,37 @@ const MarketTopDetailsOpen: React.FC<Props> = (props: Props) => {
     },
   ]
 
+  const nativeAssetAddress = getNativeAsset(context.networkId).address.toLowerCase()
+  const wrapTokenAddress = getWrapToken(context.networkId).address.toLowerCase()
+
   const onChangeMarketCurrency = (currency: Token | null) => {
     if (currency) {
-      const selectedMarket = marketsRelatedQuestion.find(e => e.collateralToken === currency.address.toLowerCase())
+      const selectedMarket = marketsRelatedQuestion.find(element => {
+        const collateralToken =
+          element.collateralToken === wrapTokenAddress ? nativeAssetAddress : element.collateralToken
+        return collateralToken === currency.address.toLowerCase()
+      })
       if (selectedMarket && selectedMarket.collateralToken !== collateral.address) {
         history.replace(`/${selectedMarket.id}`)
       }
     }
   }
 
+  const filter = marketsRelatedQuestion.map(({ collateralToken }) =>
```

Comment:
This logic is being reused in a bunch of places, could move this to a util function.

Reply:
great point!